src/Recipe_Main.tsx

```diff
@@ -0,0 +1,157 @@
+    };
+  }
+
+  const data = (await response.json()) as TData;
+  return {
+    type: 'success',
+    data,
+  };
+};
+
+const RecipeMain = () => {
+  const [recipes, setRecipes] = useState<Recipe[]>([]);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [responseMessage, setResponseMessage] = useState('');
+  const recipesPerPage = 6;
+  const LastRecipe = currentPage * recipesPerPage;
+  const FirstRecipe = LastRecipe - recipesPerPage;
+  const currentPageRecipes = recipes.slice(FirstRecipe, LastRecipe);
+  const totalPages = Math.ceil(recipes.length / recipesPerPage);
+  const navigate = useNavigate();
+
+  const fetchRecipe = () => {
+    api<{
+      recipes: Recipe[];
+      total: number;
+      skip: number;
+      limit: number;
+    }>({
+      path: 'recipes',
+      method: 'GET',
+    })
+      .then((response) => {
+        if (response.type === 'error') {
+          setResponseMessage(response.message);
+          return;
+        }
+        setRecipes(response.data.recipes);
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  };
+
+  useEffect(() => {
+    fetchRecipe();
+  }, []);
+
+  return (
+    <div>
+      <header className={styles.title_container}>👩‍🍳 천개의 레시피 🥘</header>
+      <span>{responseMessage}</span>
+      <div className={styles.recipe_main_container}>
+        <div className={styles.recipe_list}>
+          {currentPageRecipes.map((recipe) => (
+            <div
+              className={styles.recipe_container}
+              key={recipe.id}
+              onClick={() => void navigate(`/recipe/${recipe.id}`)}
+            >
+              <div className={styles.recipe_img}>
+                <img
+                  src={recipe.image}
+                  alt="레시피 이미지"
+                />
+              </div>
+              <div className={styles.recipe_info_container}>
+                <div className={styles.recipe_title_top}>
+                  <div className={styles.recipe_name}>{recipe.name}</div>
+                  <div className={styles.recipe_difficulty}>
+                    난이도: {recipe.difficulty}
+                  </div>
+                </div>
+                <div className={styles.recipe_title_bottom}>
+                  {recipe.tags.map((tag, index) => (
+                    <div
+                      key={index}
+                      className={styles.recipe_tag}
+                    >
+                      {tag}
+                    </div>
+                  ))}
+                </div>
+              </div>
+            </div>
+          ))}
+        </div>
+        <div className={styles.pagination}>
+          {Array.from({ length: totalPages }, (_, index) => (
+            <button
+              key={index + 1}
+              onClick={() => setCurrentPage(index + 1)}
+              className={currentPage === index + 1 ? styles.active : ''}
+            >
+              {index + 1}
+            </button>
+          ))}
+        </div>
```

Comment:
페이지네이션 관련된 컴포넌트나 함수들은 다른 페이지에서도 활용될 가능성이 있습니다. 이러한 컴포넌트들은 공용 컴포넌트로 관리하셔도 좋을 거 같습니다! (@/components/pagination)

Reply:
i라는 변수명이 살짝 비직관적이기는 해서 이런 식으로 바꾸는 건 어떨까요?
```suggestion
        <div className={styles.pagination}>
          {Array.from({ length: totalPages }, (_, i) => {
            const pageOfButton = i + 1
            return (<button
              className="pagination"
              key={pageOfButton}
              onClick={() => goToPage(rageOfButton)}
              style={{ fontWeight: currentPage === pageOfButton ? 'bold' : 'normal' }}
            >
              {pageOfButton}
            </button>)
          }
        </div>
```